tests/test_audio_signal_arithmetic.py

```diff
@@ -408,6 +408,27 @@ def test_complex_real_multiplication():
                         atol=1e-15)
 
 
+def test_complex_real_multiplication_freq():
+    # only test one case - everything else is tested below
+    x = Signal([1, 0, 0], 44100)
+    y = Signal([0, 1, 0], 44100, is_complex=True)
+    z = pf.multiply((x, y), 'freq')
+
+    # check result
+    x.complex = True
+    ref = x.freq * y.freq
+    npt.assert_allclose(z.freq, ref, atol=1e-15)
+
+    x = Signal([1, 0, 0], 44100, is_complex=True)
+    y = Signal([0, 1, 0], 44100)
+    z = pf.multiply((x, y), 'freq')
+
+    # check result
+    y.complex = True
```

Comment:
Can you add `assert y.complex == False` as a check to make clear that the input data is not changes